internal/component/kafka/consumer.go

```diff
@@ -188,7 +188,13 @@ func (consumer *consumer) doCallback(session sarama.ConsumerGroupSession, messag
 		Topic: message.Topic,
 		Data:  message.Value,
 	}
-
+	// This is true only when headers are set (Kafka > 0.11)
+	if message.Headers != nil && len(message.Headers) > 0 {
```

Comment:
Nit:

```suggestion
	if len(message.Headers) > 0 {
```

(if `message.Headers` is `nil`, then its length is 0)